client-core/src/wallet/syncer.rs

```diff
@@ -750,6 +865,56 @@ fn filter_incomming_staking_transactions<'a>(
     Ok(Default::default())
 }
 
+fn make_light_client_instance(
+    peer_id: PeerId,
+    addr: tendermint::net::Address,
+    db_path: impl AsRef<Path>,
+) -> Instance {
+    let mut peer_map = HashMap::new();
+    peer_map.insert(peer_id, addr);
+
+    let timeout = Duration::from_secs(10);
+    let io = ProdIo::new(peer_map, Some(timeout));
+
+    let db = sled::open(db_path).unwrap_or_else(|e| {
+        println!("[ error ] could not open database: {}", e);
+        std::process::exit(1);
+    });
+
+    let mut light_store = SledStore::new(db);
+
+    if light_store.latest(Status::Verified).is_none() {
+        // FIXME trust height 1 automatically
+        let trusted_state = io
+            .fetch_light_block(peer_id, AtHeight::At(1))
+            .unwrap_or_else(|e| {
+                println!("[ error ] could not retrieve trusted header: {}", e);
+                std::process::exit(1);
+            });
+        light_store.insert(trusted_state, Status::Verified);
+    }
+    let state = State {
+        light_store: Box::new(light_store),
+        verification_trace: HashMap::new(),
+    };
+
+    let options = light_client::Options {
+        trust_threshold: TrustThreshold {
+            numerator: 1,
+            denominator: 3,
+        },
+        trusting_period: Duration::from_secs(36000),
+        clock_drift: Duration::from_secs(1),
```

Comment:
any comment on these options? I guess threshold of 1/3 is natural.
for the period -- should it be equal to the evidence time / unbonding time?

Reply:
added some comments and links